### PR TITLE
Rearrange the 5.0 what's new page

### PR DESCRIPTION
**Pull request by roji**
This rearranges features in the "what's new" page to no longer be grouped by preview, and links to up-to-date docs which have been added since. Major/significant features have their own sections (though reduced compared to before), all other features are listed as one-liners under "other features".

I've exercised some subjective judgement in selecting which features to classify as major and which not to, hope nobody gets hurt :rofl: Obviously this is a draft and feedback is welcome :)

### CODE REVIEW

File: .markdownlint.json [hunk 1]
```diff
@@ -10,5 +10,6 @@
     "MD036": false,
     "MD046": {
         "style": "fenced"
-    }
+    },
+    "DOCSMD010": false
```
Comment (roji) [inline .markdownlint.json:14]: This suppresses the linter warning on `[foo](xref:bar#baz)` - the linker doesn't like the hash sign and suggests %23 instead, but that doesn't get picked up properly by the doc system...

File: entity-framework/toc.yml [hunk 2]
```diff
@@ -20,6 +20,8 @@
     items:
     - name: Welcome!
       href: core/index.md
+    - name: "What's new in EF Core 5.0"
+      href: core/what-is-new/ef-core-5.0/whatsnew.md
```
Comment (smitpatel) [inline entity-framework/toc.yml:24]:
```suggestion
      href: core/what-is-new/ef-core-5.0/whatsnew
```

Comment (smitpatel) [reply entity-framework/toc.yml:24]: Or may need `/ef/` prefix

Comment (roji) [reply entity-framework/toc.yml:24]: Not pushing back, but it seems to work as-is with the extension (and all other entries also have the extension)?

Comment (smitpatel) [reply entity-framework/toc.yml:24]: The subtle difference is this node is just a link to another page inside docset. That page contains actual doc content. When referencing doc content .md is required to inform this is the markdown file. For link, we can use uid directly.

Comment (roji) [reply entity-framework/toc.yml:24]: I have no idea what you just said so I'll just commit it!

Comment (smitpatel) [reply entity-framework/toc.yml]: It gave warning. 😞 
Need to put /ef/core/what-is-new/ef-core-5.0/whatsnew

Comment (roji) [reply entity-framework/toc.yml]: Done, seems to pass with no warnings

File: .markdownlint.json (hunk 1 above)
Comment (ajcvickers) [reply .markdownlint.json:14]: Ran into this yesterday.

File: entity-framework/core/what-is-new/ef-core-5.0/whatsnew.md [hunk 3]
```diff
@@ -8,13 +8,11 @@ uid: core/what-is-new/ef-core-5.0/whatsnew
 
 # What's New in EF Core 5.0
 
-All features planned for EF Core 5.0 have now been completed. This page contains an overview of interesting changes introduced in each preview.
+The following list includes the major new features in EF Core 5.0. For the full list of issues in the release, see our [issue tracker](https://github.com/dotnet/efcore/issues?q=is%3Aissue+milestone%3A5.0.0).
 
-This page does not duplicate the [plan for EF Core 5.0](xref:core/what-is-new/ef-core-5.0/plan). The plan describes the overall themes for EF Core 5.0, including everything we are planning to include before shipping the final release.
+As a major release, EF Core 5.0 also contains several [breaking changes](xref:core/what-is-new/ef-core-5.0/breaking-changes), which are API improvements that may have negative impact on existing applications.
```
Comment (ajcvickers) [inline entity-framework/core/what-is-new/ef-core-5.0/whatsnew.md:13]: Suggest: API improvement or behavior changes

**Review by ajcvickers — APPROVED** [review]

**Pull request merged**